### PR TITLE
feat: add dependencies stage

### LINKED ISSUE #3226: Show a log stage for `dependencies`
**Describe the bug**
As a follow-up of [this comment](https://github.com/okteto/okteto/pull/3219#issuecomment-1313390902) we aren't providing a log stage for deploying dependencies. This is specially tricky when using `wait`, since an error in the deployment of the dependency will be shown as an internal server error.

**Expected behavior**
Let's have a single stage for deploying dependencies, calle "Deploying dependencies".
The output will "Dependency scheduled for deployment..." if `wait` is false. And `Wating for dependency to be deployed...` + `Dependency successfully deployed` if `wait` is true. We don't want to stream the output of the dependency here because it's already available in the gitDeploy object of the dependency.

@rlamana ^^

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

# Proposed changes

Fixes #3226 

Show a new stage while deploying a depdency

Captura de Pantalla 2022-11-28 a las 17 38 13

Captura de Pantalla 2022-11-29 a las 16 27 39

Captura de Pantalla 2022-11-29 a las 16 33 21

Edit: Add more screenshoots

### CODE REVIEW

**Comment by ifbyol** [comment]
@jLopezbarb Can you include an screenshot in case some of the dependencies fail, pleaase?

Can you also include a screenshot when the pipeline is deployed (in the current screenshot it only says that the deploy is skipped)?

File: cmd/deploy/deploy.go [hunk 1]
```diff
@@ -366,6 +367,7 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 			return err
 		}
 	}
+	oktetoLog.SetStage("")
```
Comment (teresaromero) [inline cmd/deploy/deploy.go:370]: what if there is no dependency deployment, this will reset the stage from the previous one?

Comment (ifbyol) [reply cmd/deploy/deploy.go:370]: It was already clean in line 329

**Comment by jLopezbarb** [comment]
> @jLopezbarb Can you include an screenshot in case some of the dependencies fail, pleaase?
> 
> Can you also include a screenshot when the pipeline is deployed (in the current screenshot it only says that the deploy is skipped)?

Edited the first message. In the failed one it was not storing all the logs since it was not streaming them

**Review by ifbyol — APPROVED** [review]
It looks good to me! @pchico83 @rlamana Can you check the screenshots to see if it looks ok?

**Review by pchico83 — APPROVED** [review]

**Comment by rlamana** [comment]
Looks good to me to 🚀

**Review by rlamana — APPROVED** [review]

**Pull request merged**